sync/task_pool.go

```diff
@@ -267,3 +221,66 @@ func (p *TaskPool) Close() {
 		close(p.qArray[i])
 	}
 }
+
+/////////////////////////////////////////
+// Task Pool Simple
+/////////////////////////////////////////
+type taskPoolSimple struct {
+	work chan task
+	sem  chan struct{}
+}
+
+// NewTaskPoolSimple build a simple task pool
+func NewTaskPoolSimple(size int) GenericTaskPool {
+	if size < 1 {
+		size = runtime.NumCPU() * 100
+	}
+	return &taskPoolSimple{
+		work: make(chan task),
+		sem:  make(chan struct{}, size),
+	}
+}
+
+func (p *taskPoolSimple) AddTask(t task) bool {
+	select {
+	case p.work <- t:
+	case p.sem <- struct{}{}:
+		go p.worker(t)
+	}
+	return true
+}
+
+func (p *taskPoolSimple) AddTaskAlways(t task) {
+	select {
+	case p.work <- t:
+		return
+	default:
+	}
+	select {
+	case p.work <- t:
+	case p.sem <- struct{}{}:
+		go p.worker(t)
+	default:
+		goSafely(t)
+	}
+}
+
+func (p *taskPoolSimple) worker(t task) {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Fprintf(os.Stderr, "%s goroutine panic: %v\n%s\n",
+				time.Now(), r, string(debug.Stack()))
+		}
+		<-p.sem
+	}()
+	for {
+		t()
+		t = <-p.work
+	}
+}
+
+func (p *taskPoolSimple) Close() {}
```

Comment:
when to use the simple task pool? why not implement the Close() func?

Reply:
done